src/Umbraco.Cms.ManagementApi/Controllers/FolderTreeControllerBase.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Models.Entities;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.ManagementApi.ViewModels.Tree;
+
+namespace Umbraco.Cms.ManagementApi.Controllers;
+
+public abstract class FolderTreeControllerBase<TItem> : TreeControllerBase<TItem>
+    where TItem : FolderTreeItemViewModel, new()
+{
+    private readonly Guid _folderObjectTypeId;
+    private bool _foldersOnly;
+
+    protected FolderTreeControllerBase(IEntityService entityService)
+        : base(entityService) =>
+        // ReSharper disable once VirtualMemberCallInConstructor
+        _folderObjectTypeId = FolderObjectType.GetGuid();
+
+    protected abstract UmbracoObjectTypes FolderObjectType { get; }
+
+    protected async Task<ActionResult<PagedResult<TItem>>> GetRoot(long pageNumber, int pageSize, bool foldersOnly)
+    {
+        _foldersOnly = foldersOnly;
```

Comment:
This `_foldersOnly` field caused me a bit of confusion initially, since it's not immediately obvious how it's used since you have to go to `GetRoot -> GetPagedRootEntities` which you then have to realize is overridden to know how/why this is used, so I think it's a good idea to add a clarifying comment

Reply:
Will add comments :)